Revise the Query API to be more inline with the other platforms

pasin opened the pull request:
* Changed CBLQuery_New to CBLDatabase_CreateQuery()
* Changed CBLValueIndex to CBLValueIndexConfiguration
* Changed CBLFullTextIndex to CBLFullTextIndexConfiguration
* ~~Renamed kCBLN1QLLanguage to kCBLStringLanguage as N1QL term should not be used in the API.~~
* Updated Lite-Core and fixed build issue in creating index code.
* Fixed race condition in Query Listener Test.

jianminzhao commented on include/cbl/CBLQuery.h:
```diff
@@ -41,8 +41,8 @@ CBL_CAPI_BEGIN
-    kCBLJSONLanguage,   ///< [JSON query schema](https://github.com/couchbase/couchbase-lite-core/wiki/JSON-Query-Schema)
-    kCBLN1QLLanguage    ///< [N1QL syntax](https://docs.couchbase.com/server/6.0/n1ql/n1ql-language-reference/index.html)
+    kCBLJSONLanguage,       ///< [JSON query schema](https://github.com/couchbase/couchbase-lite-core/wiki/JSON-Query-Schema)
+    kCBLStringLanguage      ///< [N1QL syntax](https://docs.couchbase.com/server/6.0/n1ql/n1ql-language-reference/index.html)
```
The name is little odd to me. It makes me wonder if kCBLJSONLanguage is a graphical language. If avoiding branding name (n1ql or sql), how about CBLSequelLanguage?

pasin replied:
I think SequelLanguage is the same as SQL when pronouncing it?

jianminzhao replied:
Yes, but it avoids the connotation of proper noun of SQL

snej replied:
Why would we not use "N1QL" in the API?! It _is_ N1QL, with a few differences. (And Server already has at least two flavors of N1QL that aren't identical.) I can't think of anything else we could call it that would make sense.

pasin replied:
I will revert the change back to use N1QL. I personally agreed with you, and I have just started a separate discussion to insist to use N1QL as a name in the language options here. Once that is finalized and if the change needs to be made, the separate PR will be created.

pasin replied:
OK, I reverted back to use N1QL until it's really finalized.

pasin commented:
@jianminzhao @snej @borrrden I have updated the PR as follows:
* Revert back to use N1QL
* Updated to use the latest dev for getting the change to support creating indices using N1QL (Lite-Core API Change)
* Fixed a race condition in Query Listener test

Please (re-)review the PR. Thanks!
